File: src/jsMain/kotlin/com/scoreo/infrastructure/LocalStorageMatchDraftRepository.kt
package com.scoreo.infrastructure

import com.scoreo.domain.model.MatchDraft
import com.scoreo.domain.port.MatchDraftRepository
import kotlinx.browser.localStorage
import kotlinx.serialization.encodeToString

private const val KEY = "scoreo_match_draft"

class LocalStorageMatchDraftRepository : MatchDraftRepository {
    override fun save(draft: MatchDraft) {
        localStorage.setItem(KEY, scoreoJson.encodeToString(draft))
    }

    override fun load(): MatchDraft? {
        val raw = localStorage.getItem(KEY) ?: return null
        return try {
            scoreoJson.decodeFromString<MatchDraft>(raw)
        } catch (e: Exception) {
            null // Graceful fallback on deserialization error
        }
    }

    override fun clear() {
        localStorage.removeItem(KEY)
    }
}

File: apps/scoreo/src/infrastructure/localStorage/localStorageMatchDraftRepository.ts
import type { MatchDraft } from '../../domain/model/matchDraft'
import { MatchDraftSchema } from '../../domain/model/matchDraft.schema'
import type { MatchDraftRepository } from '../../domain/port/matchDraftRepository'

const KEY = 'scoreo_match_draft'

export class LocalStorageMatchDraftRepository implements MatchDraftRepository {
  save(draft: MatchDraft): void {
    localStorage.setItem(KEY, JSON.stringify(draft))
  }

  load(): MatchDraft | undefined {
    const raw = localStorage.getItem(KEY)
    if (raw === null) return undefined
    try {
      return MatchDraftSchema.parse(JSON.parse(raw))
    } catch {
      return undefined
    }
  }

  clear(): void {
    localStorage.removeItem(KEY)
  }
}
